src/agreement.rs

```diff
@@ -94,8 +94,28 @@ pub struct Algorithm {
     // `ec::AgreementAlgorithmImpl` isn't public.
     #[doc(hidden)]
     pub i: ec::AgreementAlgorithmImpl,
+    #[doc(hidden)]
+    pub id: AlgorithmID,
+}
+
+/// Key agreement algorithm types.
+#[derive(PartialEq, Eq)]
+#[allow(non_camel_case_types)]
```

Comment:
Let's put the `allow(...)` ahead of `derive(...)` in this and the other ones. No order seems better than any other, so let's things in alphabetical order so we don't have to decide another order.